operations/hpe_pdu/hpe_pdu_admin_procedures.md

```diff
@@ -0,0 +1,78 @@
+
+#### Add default user
+1. Using the **"admin"** menu (top right corner) go to **"User Accounts"**
+2. Click on the **"Add User"** button
+3. Using the form, add the _username_ and _password_ for the default river user.  Assign the role _"Administrator"_ to that user.
+
+#### Enable Outlet Control
+1. Using the **"Home"** icon (House in top right corner) go to **"Control & Manage"**
+2. Make sure **"Outlet Control Enable"** switch on the top of the page is selected (green)
+
+#### Updating HPE PDU Firmware
+##### Check Firmware Version
+1. Connect to the HPE PDU Web Interface (See [Connecting to HPE PDU Web Interface](#connecting-to-hpe-pdu-web-interface)) and login as admin
+2. Check which version of firmware is installed by selecting the **"Home"** icon (House in the top right corner) go to **"Identification"**
+3. The _"Version"_ will be displayed.  If the version is less than _"2.0.0.L"_ update firmware.
+
+##### Update Firmware
+1. Download the latest firmware from: [https://support.hpe.com/connect/s/search?language=en_US#q=P9S23A&t=All&sort=%40hpescuniversaldate%20descending&numberOfResults=25&f:@contenttype=[Drivers%20and%20Software]](https://support.hpe.com/connect/s/search?language=en_US#q=P9S23A&t=All&sort=%40hpescuniversaldate%20descending&numberOfResults=25&f:@contenttype=[Drivers%20and%20Software])
+2. This will download an ".exe" file which is a self extracting zip file.  If using a windows system, run the .exe file to extract the files, or use an unzip program on the file.  One of the files extracted will be named **"HPE.FW"**, that is the firmware file you will need for uploading.
+3. Connect to the HPE PDU Web Interface (See [Connecting to HPE PDU Web Interface](#connecting-to-hpe-pdu-web-interface)) and login as admin
+4. Using the **"Settings"** Icon (gear in computer monitor in top right corner) go to **"System Management"**
+5. Click the **"Update FIrmware"** button
+6. Click **"Choose File"** and select the **"HPE.FW"** file downloaded
+7. Click **"Upload"** button
+8. Firmware will be updated and the PDU management processor will restart.
+
+#### Changing HPE PDU User Passwords
+1. Connect to the HPE PDU Web Interface (See [Connecting to HPE PDU Web Interface](#connecting-to-hpe-pdu-web-interface)) and login as admin
+2. Using the **"admin"** menu (top right corner) go to **"User Accounts"**
+3. Click on the edit icon (pencil) next to the user
+4. Enter new password and make any other changes for that user account and click the "Save" button
+
+#### Discovery of HPE PDU after upgrading CSM
+**This procedure is only needed when upgrading CSM, not performing a fresh install.  This procedure should be run after CSM has been fully upgraded including the discovery job.**
+1. Find the list of PDU MAC address:\
+`# cray hsm inventory ethernetInterfaces list --type CabinetPDUController`\
+This will return a list of PDU ethernet interfaces.  Each interface must be deleted.
+

```

Comment:
We should also add a step to clean up the RTS area of Vault, as after the upgrade RTS will think these are still ServerTech pdus:


````
2.  In CSM 1.0 and before the hms-discovery job and Redfish Translation Service (RTS) treated all PDUs as if were made by ServerTech, and after the upgrade to CSM 1.2 RTS will still think the HPE PDUs in the system are ServerTech PDUs. These erroneous HPE PDU entries for RTS need to get removed from Vault.  
    ```bash
    ncn-m001# VAULT_PASSWD=$(kubectl -n vault get secrets cray-vault-unseal-keys -o json | jq -r '.data["vault-root"]' |  base64 -d)
    ncn-m001# alias vault='kubectl -n vault exec -i cray-vault-0 -c vault -- env VAULT_TOKEN=$VAULT_PASSWD VAULT_ADDR=http://127.0.0.1:8200 VAULT_FORMAT=json vault'
    ```
    1.  For each HPE PDU Identified in the command above remove it from Vault:
        ```bash
        ncn-m001# PDU=x3000m0
        ncn-m001# vault kv delete secret/pdu-creds/$PDU
        ```
    2.  Restart the Redfish Translation Service (RTS):
        ```bash
        ncn-m001# kubectl -n services rollout restart deployment cray-hms-rts
        ncn-m001# kubectl -n services rollout status deployment cray-hms-rts
        ```
````

Reply:
Added as step number 1, let me know if it needs to move.